src/Naderman/Composer/AWS/AwsClient.php

```diff
@@ -125,39 +161,89 @@ public function determineBucketAndKey($url)
         }
         $bucket = array_shift($parts);
         $key = implode('/', $parts);
+        if (!$key) {
+            $key = '/';
+        }
         return array($bucket, $key);
     }
 
     /**
+     * This method reads AWS config and credentials and create s3 client
+     * Behaviour aims to mimic region setup as it is for credentials:
+     * http://docs.aws.amazon.com/aws-sdk-php/v3/guide/getting-started/basic-usage.html#creating-a-client
+     * Which is the following (stopping at the first successful case):
+     * 1) read region from config parameter
+     * 2) read region from environment variables
+     * 3) read region from profile config file
+     * 4) set region US Standard region
+     * 
      * @param \Composer\Config $config
      *
      * @return \Aws\S3\S3Client
      */
-    public static function s3factory(Config $config)
+    public function s3factory(Config $config)
     {
-        $s3config = array(
-            'default_cache_config'  => '',
-            'certificate_authority' => '',
-        );
-
-        /**
-         * If these are not set and we happen to have an IAM profile, it will still work.
-         */
-        if (($composerAws = $config->get('amazon-aws'))) {
-            $s3config = array_merge($s3config, $composerAws);
-        } else {
+        if (is_null($this->client)) {
+            $s3config = array(
+                'version' => 'latest'
+            );
+    
             /**
-             *  Attempt to fall back to environment variables.
+             * If these are not set and we happen to have an IAM profile, it will still work.
              */
-            $key = getenv('AWS_ACCESS_KEY_ID');
-            $secret = getenv('AWS_SECRET_ACCESS_KEY');
-
-            if (false !== $key && false !== $secret) {
-                $s3config['key'] = $key;
-                $s3config['secret'] = $secret;
+            if (($composerAws = $config->get('amazon-aws'))) {
+                $s3config = array_merge($s3config, $composerAws);
```

Comment:
Do you think you can just stick in the "credentials" key here if needed, so that the config remains backward compatible?